test/parallel/test-http-parser-lazy-loaded.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+// Monkey patch before requiring anything
+class DummyParser {
+  constructor(type) {
+    this.test_type = type;
+  }
+}
+DummyParser.REQUEST = Symbol();
+process.binding('http_parser').HTTPParser = DummyParser;
```

Comment:
Oh, and I think this should be using `internalBinding`, since we recently moved toward gradually deprecating all of `process.binding()`.

Reply:
I saw that and tried `internalBinding`, but it complained with `ReferenceError: internalBinding is not defined`. Do you need to do something special to access this in tests and/or user-level code?  I saw that 'http_parser' was on the 'process.binding white list' at some point, but I'm not sure if that's so that users can still get at it, or just a migration path toward deprecation.

Reply:
Yes, put this at the top of the file:

```
// Flags: --expose-internals
```

That lets you specify command line arguments that the test needs.

Then:

```
const { internalBinding } = require('internal/test/binding');
```